### PR TITLE
Ease restriction on native-sources type and container image extension

### LINKED ISSUE #24351: Cannot build native-source after adding container-image-buildpack extension.
### Describe the bug

Cannot use native-source packaging when container image extension is added to a project.

### Expected behavior

The maven pack task will pass without error.

### Actual behavior

The maven package task fails:
```
./mvnw package -DskipTests=true -Dmaven.javadoc.skip=true -Dquarkus.package.type=native-sources
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< cz.mvasek:quarkus-buildpack >---------------------
[INFO] Building quarkus-buildpack 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ quarkus-buildpack ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO] 
[INFO] --- quarkus-maven-plugin:2.7.4.Final:generate-code (default) @ quarkus-buildpack ---
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ quarkus-buildpack ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- quarkus-maven-plugin:2.7.4.Final:generate-code-tests (default) @ quarkus-buildpack ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ quarkus-buildpack ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /tmp/tmp.DgIJNqy34r/quarkus-buildpack/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ quarkus-buildpack ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ quarkus-buildpack ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ quarkus-buildpack ---
[INFO] 
[INFO] --- quarkus-maven-plugin:2.7.4.Final:build (default) @ quarkus-buildpack ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.333 s
[INFO] Finished at: 2022-03-16T14:32:59+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.7.4.Final:build (default) on project quarkus-buildpack: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.container.image.deployment.ContainerImageProcessor#failForNativeSources threw an exception: java.lang.IllegalArgumentException: The Container Image extensions are incompatible with the 'native-sources' package type.
[ERROR] 	at io.quarkus.container.image.deployment.ContainerImageProcessor.failForNativeSources(ContainerImageProcessor.java:26)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

### How to Reproduce?

1. Create simple RESTEasy app using https://code.quarkus.io/?e=resteasy
2. Add the container-image-buildpack extension `./mvnw quarkus:add-extension -Dextensions="container-image-buildpack"` to the project.
3. Try to build project with `native-source`: `./mvnw package -DskipTests=true -Dmaven.javadoc.skip=true -Dquarkus.package.type=native-sources`

The build fails. Maybe I could understand if that happended if `quarkus.container-image.build=true`, but it happens even when the container image build is not in use.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
The build now only fails when the package type is set to native-sources
and one of the container image extensions would actually perform a
container build.

Fixes: #24351

### CODE REVIEW

**Comment by geoand** [comment]
@iocanel the first commit probably needs a careful check

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java [hunk 1]
```diff
@@ -73,12 +75,9 @@ public void dockerBuildFromJar(DockerConfig dockerConfig,
...
             JarBuildItem jar) {
 
-        if (containerImageConfig.isBuildExplicitlyDisabled() && !containerImageConfig.isPushExplicitlyEnabled()) {
-            return;
-        }
-
-        if (!containerImageConfig.isBuildExplicitlyEnabled() && !containerImageConfig.isPushExplicitlyEnabled()
-                && !buildRequest.isPresent() && !pushRequest.isPresent()) {
+        Boolean buildContainerImage = buildContainerImageNeeded(containerImageConfig, buildRequest);
+        Boolean pushContainerImage = pushContainerImageNeeded(containerImageConfig, pushRequest);
```
Comment (gastaldi) [inline extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java:79]:
```suggestion
        boolean buildContainerImage = buildContainerImageNeeded(containerImageConfig, buildRequest);
        boolean pushContainerImage = pushContainerImageNeeded(containerImageConfig, pushRequest);
```

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java [hunk 2]
```diff
@@ -125,12 +124,9 @@ public void dockerBuildFromNativeImage(DockerConfig dockerConfig,
...
             NativeImageBuildItem nativeImage) {
 
-        if (containerImageConfig.isBuildExplicitlyDisabled() && !containerImageConfig.isPushExplicitlyEnabled()) {
-            return;
-        }
-
-        if (!containerImageConfig.isBuildExplicitlyEnabled() && !containerImageConfig.isPushExplicitlyEnabled()
-                && !buildRequest.isPresent() && !pushRequest.isPresent()) {
+        Boolean buildContainerImage = buildContainerImageNeeded(containerImageConfig, buildRequest);
+        Boolean pushContainerImage = pushContainerImageNeeded(containerImageConfig, pushRequest);
```
Comment (gastaldi) [inline extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java:128]:
```suggestion
        boolean buildContainerImage = buildContainerImageNeeded(containerImageConfig, buildRequest);
        boolean pushContainerImage = pushContainerImageNeeded(containerImageConfig, pushRequest);
```

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by geoand** [comment]
@iocanel mind taking a look at this one please?

**Review by iocanel — APPROVED** [review]
LGTM

**Pull request merged**